domain/removal/state/controller/model.go

```diff
@@ -97,6 +99,118 @@ func (st *State) GetModelLife(ctx context.Context, mUUID string) (life.Life, err
 	return life, errors.Capture(err)
 }
 
+// MarkModelAsDead marks the model with the input UUID as dead.
+// If there are model dependents, then this will return an error.
+func (st *State) MarkModelAsDead(ctx context.Context, mUUID string) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	modelUUID := entityUUID{UUID: mUUID}
+	updateStmt, err := st.Prepare(`
+UPDATE model
+SET    life_id = 2
+WHERE  uuid = $entityUUID.uuid
+AND    life_id = 1`, modelUUID)
+	if err != nil {
+		return errors.Errorf("preparing model life update: %w", err)
+	}
+	return errors.Capture(db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		if l, err := st.getModelLife(ctx, tx, mUUID); err != nil {
```

Comment:
You could save yourself a query on the sunny day path by just running the update, and querying if 0 rows were updated.

Reply:
I think I wanted a better error message. Otherwise I agree.